app/uk/gov/nationalarchives/omega/editorial/services/jms/ResponseBuilder.scala

```diff
@@ -44,26 +44,50 @@ class ResponseBuilder[F[_] : ResponseBuilder.ME : Logger] {
 
   def createResponseText(jmsMessage: JmsMessage): F[String] =
     jmsMessage.getStringProperty(sidHeaderKey) match {
-      case Some(sidValue) if sidValue == sid1 =>
-        logger.info(s"got a message with sid header '$sidValue', trying to parse payload") *>
-          getEditSet(jmsMessage).flatMap { editSet =>
-            logger.info("parsed payload, creating edit set response") *>
-              me.pure(Json.toJson(editSet).toString)
-          }
-
+      case Some(sidValue) if sidValue == SID.GetEditSet =>
+        handleRequestForGettingAnEditSet(jmsMessage, sidValue)
+      case Some(sidValue) if sidValue == ResponseBuilder.SID.GetEditSetRecord =>
+        handleRequestForGettingAnEditSetRecord(jmsMessage)
       case unknown =>
-        logger.error(s"Expected a SID value of $sid1, but got $unknown, failing") *>
-          me.raiseError(new NotImplementedError(s"unhandled case for SID $unknown"))
-
+        onUnhandledCase(s"SID is unrecognised: [$unknown]")
     }
 
+  private def handleRequestForGettingAnEditSet(jmsMessage: JmsMessage, sidValue: String): F[String] =
```

Comment:
I think the naming of these functions is a little unwieldy. Could we cut them down to something like `handleGetEditSet` and `handleGetEditSetRecord`? Given they take a `JmsMessage` as a parameter I would assume the `Request` bit is implied.

Reply:
Agreed.

Reply:
Changed.

Reply:
Marking resolved, as you've approved, @felixgb .